### PR TITLE
fix(meta): effective ip

### LINKED ISSUE #867: [Improve] Check if the IP address of the cluster node is valid(POC)
Sometimes different nodes specify the same ip address, 
which affects queries and writes. 
We should detect it and give a hint.

### PR DESCRIPTION
**Pull request by yuluo-zy**
# Which issue does this PR close?

Closes #867

# Rationale for this change

Check if the IP address of the cluster node is valid.
If we add the submitted ip for the same cluster, we should output the error log.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by roseboy-liu — APPROVED** [review]

**Comment by ZuoTiJia** [comment]
Thanks for your PR, could you use `make fmt` to format the code before committing?

File: meta/src/store/state_machine.rs [hunk 1]
```diff
@@ -647,6 +647,12 @@ impl StateMachine {
     }
 
     fn process_add_date_node(&self, cluster: &str, node: &NodeInfo) -> CommandResp {
+        for (_, value) in children_data::<NodeInfo>(&KeyPath::data_nodes(cluster), self.db.clone()).iter() {
+            // Check whether the ip of the same cluster has been added
+            if value.id == node.id && value.http_addr == node.http_addr {
```
Comment (ZuoTiJia) [inline meta/src/store/state_machine.rs:652]: The relationship between node id and httpaddr is one-to-one, the logic of the if statement should be or, and tcpaddr should also be added.

Comment (yuluo-zy) [reply meta/src/store/state_machine.rs:652]: Thanks, I will fix these issues.

**Comment by yuluo-zy** [comment]
Please help to review, currently check whether the address of http and tcp has been added, if it has been added, output error message and related existing node information @ZuoTiJia

**Review by ZuoTiJia — APPROVED** [review]

File: meta/src/store/state_machine.rs [hunk 2]
```diff
@@ -647,6 +647,14 @@ impl StateMachine {
     }
 
     fn process_add_date_node(&self, cluster: &str, node: &NodeInfo) -> CommandResp {
+        for (_, value) in
+            children_data::<NodeInfo>(&KeyPath::data_nodes(cluster), self.db.clone()).iter()
+        {
+            // Check whether the ip of the same cluster has been added
+            if value.http_addr == node.http_addr || value.tcp_addr == node.tcp_addr {
```
Comment (ZuoTiJia) [inline meta/src/store/state_machine.rs:654]: node_id should also be checked.

Comment (yuluo-zy) [reply meta/src/store/state_machine.rs:654]: If you check the node_id, will you output an error when the node_id is duplicated?

**Review by ZuoTiJia — APPROVED** [review]

**Pull request merged**